src/runtime/HalideRuntimeHexagonDma.h

```diff
@@ -7,6 +7,27 @@
 extern "C" {
 #endif
 
+/*!
+ * Image Formats to prepare the application for DMA Transfer
+ */
+typedef enum __attribute__((aligned(4))) image_fmt {
```

Comment:
I don't think you need the alignment attribute (because this is not used to declare any struct members, and it should not be), and it's potentially a source of confusion in the public API, so I'd remove it.

Also, the convention for enums (from looking at https://github.com/halide/Halide/blob/master/src/runtime/HalideRuntime.h) is:
- The name should be `halide_`, probably `halide_hexagon_`. It's kind of verbose but I think it would be better.
- The suffix of the enum type should be _t (rather than prefixed with t_)
- There seems to be some existing confusion in our headers about `typedef enum { ... } blah_t` vs. `typedef enum blah { ... } blah_t`. I would lean towards the former (just the global identifier, no enum identifier).